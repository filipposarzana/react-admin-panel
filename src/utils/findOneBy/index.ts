import { User, users } from '~/db'

type Key = Extract<keyof User, 'name' | 'uuid'>

const findOneBy = <K extends Key>(prop: K) => (value: User[K]) =>
  users.find((user) => user[prop].toLowerCase() === value.toLowerCase())

export const findOneByName = findOneBy('name')

export const findOneByUuid = findOneBy('uuid')
