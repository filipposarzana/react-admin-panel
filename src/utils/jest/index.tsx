import { render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { ReactElement } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { StyleSheetManager } from 'styled-components'

export { act, fireEvent, waitFor } from '@testing-library/react'

export const renderer = {
  deep: (node: ReactElement) => {
    const r = render(<StyleSheetManager disableVendorPrefixes>{node}</StyleSheetManager>)

    return {
      ...r,
      output: r.container.firstChild,
    }
  },
  hook: <T extends unknown, R extends unknown>(hookFunction: (props: T) => R) => {
    const r = renderHook(hookFunction)

    return r
  },
  shallow: (node: ReactElement) => {
    const r = createRenderer()

    r.render(node)

    return {
      instance: r.getMountedInstance(),
      output: r.getRenderOutput(),
      unmount: r.unmount,
    }
  },
}
