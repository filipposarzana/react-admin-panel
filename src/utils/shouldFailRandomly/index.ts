export const shouldFailRandomly = () => Math.round(Math.random()) === 0
