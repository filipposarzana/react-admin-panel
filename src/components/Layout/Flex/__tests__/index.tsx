import { renderer } from '~/utils/jest'
import { Flex, StyleProps } from '..'

const fixtures: [string, StyleProps][] = [
  ['matches snapshot with default props', {}],
  ['matches snapshot with background props', { background: 'gray300' }],
  ['matches snapshot with border-radius props', { borderRadius: 4 }],
  ['matches snapshot with disabled props', { disabled: true }],
  ['matches snapshot with flex props', { align: undefined }],
  ['matches snapshot with padding props', { p: 4, px: 4, py: 4, pt: 4, pb: 4, pr: 4, pl: 4 }],
]

describe('Flex', () => {
  fixtures.forEach(([title, props]) => {
    it(title, () => {
      const { output } = renderer.deep(<Flex {...props} />)

      expect(output).toMatchSnapshot()
    })
  })
})
