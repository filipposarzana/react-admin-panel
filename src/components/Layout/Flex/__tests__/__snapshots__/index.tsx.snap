// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex matches snapshot with background props 1`] = `
.c0 {
  background-color: rgb(118,118,118);
  display: flex;
  min-height: 0;
  min-width: 0;
  align-items: stretch;
  flex-basis: auto;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-start;
  flex-shrink: 0;
  flex-wrap: nowrap;
}

<div
  class="c0"
  direction="column"
  wrap="nowrap"
/>
`;

exports[`Flex matches snapshot with border-radius props 1`] = `
.c0 {
  border-radius: 4px;
  overflow: hidden;
  display: flex;
  min-height: 0;
  min-width: 0;
  align-items: stretch;
  flex-basis: auto;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-start;
  flex-shrink: 0;
  flex-wrap: nowrap;
}

<div
  class="c0"
  direction="column"
  wrap="nowrap"
/>
`;

exports[`Flex matches snapshot with default props 1`] = `
.c0 {
  display: flex;
  min-height: 0;
  min-width: 0;
  align-items: stretch;
  flex-basis: auto;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-start;
  flex-shrink: 0;
  flex-wrap: nowrap;
}

<div
  class="c0"
  direction="column"
  wrap="nowrap"
/>
`;

exports[`Flex matches snapshot with disabled props 1`] = `
.c0 {
  pointer-events: none;
  display: flex;
  min-height: 0;
  min-width: 0;
  align-items: stretch;
  flex-basis: auto;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-start;
  flex-shrink: 0;
  flex-wrap: nowrap;
}

<div
  class="c0"
  direction="column"
  disabled=""
  wrap="nowrap"
/>
`;

exports[`Flex matches snapshot with flex props 1`] = `
.c0 {
  display: flex;
  min-height: 0;
  min-width: 0;
  flex-basis: auto;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-start;
  flex-shrink: 0;
  flex-wrap: nowrap;
}

<div
  class="c0"
  direction="column"
  wrap="nowrap"
/>
`;

exports[`Flex matches snapshot with padding props 1`] = `
.c0 {
  display: flex;
  min-height: 0;
  min-width: 0;
  align-items: stretch;
  flex-basis: auto;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-start;
  flex-shrink: 0;
  flex-wrap: nowrap;
  padding: 4px;
  padding-left: 4px;
  padding-right: 4px;
  padding-bottom: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 4px;
}

<div
  class="c0"
  direction="column"
  wrap="nowrap"
/>
`;
